Compile tests against latest protege

rgrinberg opened the pull request:
* MetaprojectFactory is now ConfigurationManager
* Ontology has to be passed to the project when creating

The tests still don't pass but at least they compile now. For now the errors seem to be because the local test server config is faulty.

rgrinberg committed:
Compile tests against latest protege

* MetaprojectFactory is now ConfigurationManager
* Ontology has to be passed to the project when creating

bdionne commented:
so far so good. Nice to see the `RMI` cruft go away

bdionne committed:
WIP - upgrade tests

This repo has atrophied quite a bit

rgrinberg commented:
@bdionne I've updated the config to be readable. You should still be able to easily rebase this PR. I'll warn you before I push something intrusive. Also feel free to just push your fixes here directly.

rgrinberg commented:
Actually, it seems like you've done similar changes to the config so I just pulled yours over mine.

bdionne commented:
right, I just pulled the latest from the server, knowing it parses, and then started changing the code to support the newer stuff, like the need for projects to be created using the `8081` port, etc..

you'll soon run into the issue that `SessionRecorder` needs an `OWLEditKit` to instantiate one, so I imagine some sort of Mock will be needed

rgrinberg commented:
I'm seeing the following deserialization error.
```
org.protege.editor.owl.client.api.exception.ClientRequestException: Failed to read data from server (see error log for details)

	at org.protege.editor.owl.client.LocalHttpClient.retrieveServerDocumentFromServerResponse(LocalHttpClient.java:217)
	at org.protege.editor.owl.client.LocalHttpClient.postProjectToServer(LocalHttpClient.java:193)
	at org.protege.editor.owl.client.LocalHttpClient.createProject(LocalHttpClient.java:177)
	at org.protege.editor.owl.integration.CommitChangesTest.createProject(CommitChangesTest.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2624)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:3099)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:853)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:349)
	at org.protege.editor.owl.client.LocalHttpClient.retrieveServerDocumentFromServerResponse(LocalHttpClient.java:212)
	... 38 more
```

bdionne committed:
WIP - give bob workflow manager powers

NOTE: this requires change to protege-client, see upgrade-tests there for detail

rgrinberg committed:
Fix NewProjectTest

rgrinberg commented:
@bdionne is the last commit the right approach? I fixed that NewProjectTest and can fix other tests use the admin client for this erroneously.

rgrinberg committed:
Add logging to tests

rgrinberg committed:
Fix pom to run tests properly

we should set the system properties on the surefire plugin rather than the
failsafe one. As failsafe simply forks off a surefire process that doesn't
inherit the properties

rgrinberg commented:
@bdionne Pulled your latest commit

bdionne commented:
getting closer, they almost all run now for me except `shouldCommitAddition` and `createLargeProject`

rgrinberg commented:
Closing this as there's a more  up to date branch.
